Feature/22 home

5nam opened the pull request:
### PR 이후 개발해야 하는 기능들

- [x] 예외처리

> Member 조회 후 없을 경우 예외처리

- [ ] MEMBER 식별

> TOP15 에서 관심상점 여부를 나타낼 때 필요한 데이터

- [x] S3 기능 연동
- [ ] 테스트 코드

### 반영 브랜치

feature/22 home

### 변경 사항

- ReviewRepositroy 생성 -> 구현 기능 중 찾기가 필요해서
- home 페이지 데이터 반환 기능 구현
- Store Entity 에 name 필드 생긴 것 반영

### 테스트 결과

Postman 테스트 결과 현재는 이상 없습니다.

해당 PR 은 Member, Review 등이랑 많이 얽힌 부분이라 다른 기능까지 합친 후, 재테스트 후 merge 할 예정입니다.

5nam committed:
:card_file_box: 필요한 필드 추가
1. 관심상점 조회를 위한 List<HeartStores> 추가
2. 상점 이름 추가

5nam committed:
:card_file_box: Getter 어노테이션 추가

5nam committed:
:sparkles: 홈 조회 기능 구현
1. top15 : 최다 리뷰 횟수 순으로 15개의 상점 리스트
2. HOT 상점 : 랜덤으로 HOT 상점 3개 뽑아내기
3. 또또가 리뷰 : 랜덤으로 리뷰 5개 뽑아내기

5nam committed:
:bulb: Member 관련 로직 위치에 미리 주석 추가

5nam committed:
:recycle: Home 응답에 reviewLink 추가

5nam committed:
:recycle: 홈 기능 리뷰 부분 변경
1. 기존 리뷰 링크에서 받아오려던 데이터 Review, Member 데이터에서 받아오는 것으로 변경
2. 그 과정에서 Member Getter 추가

5nam committed:
:bulb: 불필요한 주석 제거

5nam committed:
:card_file_box: Store 에 name 필드 생긴 것 반영

5nam committed:
[Merge] 충돌 코드 merge
1. Member 에 기본 생성자가 필요함
- 토큰에서 받아오는 멤버 정보를 일단 대체할 객체가 필요하기 때문
- 추후에 지울 예정

YeowonKIM commented on src/main/java/com/umc/ttg/domain/store/application/StoreCommandServiceImpl.java:
```diff
@@ -56,4 +67,71 @@ public BaseResponseDto<StoreFindResponseDto> findStore(Long storeId) {
+
+        // ResponseDTO
+        HomeResponseDto homeResponseDto = HomeResponseDto.builder()
+                .top15(top15)
```
기능별로 분리해서 가독성이 좋습니다 👍

YeowonKIM commented:
기능별로 잘 구분되어 있고 가독성도 좋은 것 같아요!
충돌난 거만 확인하셔서 머지하면 좋을 것 같습니다 👍

5nam committed:
:twisted_rightwards_arrows: merge 충돌 해결

5nam committed:
:recycle: Controller 에서 Member 객체 생성하지 않고, Service 에서 생성하도록 로직 변경

5nam committed:
Merge branch 'develop' into feature/22-home

5nam committed:
:recycle: 관심상점 등록 여부 처리하는 로직 간편하게 변경
1. 기존 로직은 Store 에 HeartStores 라는 리스트를 두어서 member 객체로 조회하는 방식을 사용함
2. 변경 로직은 HeartStoreRepository 에서 Member, Store 객체로 조회 후 있으면 true, 없으면 false 를 반환하는 방식

5nam committed:
Merge branch 'develop' into feature/22-home

5nam committed:
Merge branch 'develop' into feature/22-home

5nam committed:
:twisted_rightwards_arrows: develop 머지
1. develop - 지역별 Store 조회 기능과 merge

5nam committed:
:recycle: Store 생성자 private 로 막아놓고, builder 로 변경
1. setter 를 없앰

5nam committed:
:bug: Store 생성자를 Private & Setter 없애는 리팩터링 후 menu, region, image null 로 저장되는 문제
1. 생성자 내부 로직에 반영을 안해줘서 문제 발생
-> 로직 반영하여 해결

5nam committed:
:sparkles: 메뉴별 상점 조회 기능 구현
1. 리팩터링도 동시에 진행
- 지역별 상점 조회와 메뉴별 상점 조회에 겹치는 기능이 많아서 함수로 분리
- comparator 와 paging 기능 분리
2. findAllByRegionOrderBySort 와 findAllByMenuOrderBySort 는 나눠져 있었는데, 하나의 함수로 처리할 수 있을듯 하여 getAllByObject 로 합침
- Object 로 받아서, instanceof 를 이용해 타입별(Region, Menu) 별로 처리

5nam committed:
:bug: Store 저장 시 menu, region, image null 로 저장되는 문제
1. Store 생성자를 private 로 변경하고, setter 를 없애는 리팩터링 후 생성자 내부 로직에 menu, region, image 반영하지 않아서 발생
- 반영하여 해결

5nam committed:
:sparkles: 메뉴별 상점 조회 기능 구현
1. 리팩터링도 동시에 진행
- 지역별 상점 조회와 메뉴별 상점 조회에 겹치는 기능이 많아서 함수로 분리
- comparator 와 paging 기능 분리
2. findAllByRegionOrderBySort 와 findAllByMenuOrderBySort 는 나눠져 있었는데, 하나의 함수로 처리할 수 있을듯 하여 getAllByObject 로 합침
- Object 로 받아서, instanceof 를 이용해 타입별(Region, Menu) 별로 처리

5nam committed:
Merge branch 'feature/22-home' of https://github.com/Ttottoga/BE into feature/22-home

5nam committed:
:sparkles: 상점 조회 시 리뷰를 남겼던 상점인지 아닌지 여부 반환하는 기능 추가
1. submitReview 가 true -> 리뷰를 남겼던 곳, false -> 리뷰를 남기지 않았던 곳
